extensions/wikia/UserRenameTool/UserRenameToolProcess.class.php

```diff
@@ -246,197 +194,358 @@ protected function renameAccount() {
 		return false;
 	}
 
-	/**
-	 * Checks if the request provided to the constructor is valid.
-	 *
-	 * @return bool True if all prerequisites are met
-	 */
-	protected function setup() {
-		global $wgContLang, $wgCapitalLinks;
+	private function antiSpoofTest( $newUserName ) {
+		if ( class_exists( 'SpoofUser' ) ) {
+			$oNewSpoofUser = new SpoofUser( $newUserName );
+			if ( !$oNewSpoofUser->isLegal() ) {
+				$this->addWarning( wfMessage( 'userrenametool-error-antispoof-conflict', $newUserName ) );
+			}
+		} else {
+			$this->addError( wfMessage( 'userrenametool-error-antispoof-notinstalled' ) );
+		}
+	}
 
-		// Sanitize input data
-		$oldNamePar = trim( str_replace( '_', ' ', $this->mRequestData->oldUsername ) );
-		$oldTitle = Title::makeTitle( NS_USER, $oldNamePar );
+	private function phalanxTest( $userName ) {
+		$warning = RenameUserHelper::testBlock( $userName );
+		if ( !empty( $warning ) ) {
+			$this->addWarning( $warning );
+		}
+	}
 
-		// Force uppercase of newusername, otherwise wikis with wgCapitalLinks=false can create lc usernames
-		$newTitle = Title::makeTitleSafe( NS_USER, $wgContLang->ucfirst( $this->mRequestData->newUsername ) );
+	private function areUserNamesValid( $oldUserName, $newUsername ) {
+		if ( !$this->isUserNameNotEmpty(
```

Comment:
The double negative ("not is username not empty") is a little confusing.  They could be cancelled out to be:
```php
if ( $this->isUserNameEmpty ) {
```